src/Scout/ScoutEnhancer.php

```diff
@@ -145,4 +112,42 @@ function ($value) {
             );
         }
     }
+
+    protected function enhanceEloquentBuilder(): ScoutBuilder
+    {
+        if (count($this->searchArguments) > 1) {
+            throw new ScoutException('Found more than 1 argument with @search.');
+        }
+        $searchArgument = $this->searchArguments[0];
+
+        if (count($this->argumentsWithOnlyArgBuilders) > 0) {
+            throw new ScoutException('Found arg builder arguments that do not work with @search');
+        }
+
+        if (! $this->builder instanceof EloquentBuilder) {
+            throw new ScoutException('Can only get Model from \Illuminate\Database\Eloquent\Builder, got: ' . get_class($this->builder));
+        }
+        $model = $this->builder->getModel();
+
+        if (! Utils::classUsesTrait($model, Searchable::class)) {
+            throw new ScoutException('Model class ' . get_class($model) . ' does not implement trait ' . Searchable::class);
+        }
+
+        // @phpstan-ignore-next-line Can not use traits as types
+        /**
+         * @var \Illuminate\Database\Eloquent\Model&\Laravel\Scout\Searchable $model
+         */
+        $scoutBuilder = $model::search($searchArgument->toPlain());
+
+        /**
+         * We know this argument has this directive, because that is how we found it.
+         *
+         * @var \Nuwave\Lighthouse\Scout\SearchDirective $searchDirective
+         */
+        $searchDirective = $searchArgument
+            ->directives
+            ->first(Utils::instanceofMatcher(SearchDirective::class));
+
+        $searchDirective->search($scoutBuilder);
+    }
```

Comment:
Obviously I forgot : 

```suggestion
        return $scoutBuilder;
    }
```